### PR TITLE
Change the default init_dist in timeseries and issue UserWarning

### LINKED ISSUE #5744: Improve default GaussianRandomWalk and AR init distributions
When refactoring the GaussianRandomWalk to V4, we set the `init` to default to `Normal(mu, sigma)`, which does not make much sense. `mu` is the drift parameter, and says nothing about the expected starting location of the series. `sigma` probably also makes little sense, as it assumes the uncertainty about the initial location has anything to do with the innovations noise. 

Possible alternatives:
* `init = Normal.dist(0, 1)`
* `init = Flat.dist()`, which was the default in V3, but it means that prior and posterior predictive will fail by default.

We should also rename `init` to `init_dist` to emphasize this parameter must de a distribution.

https://github.com/pymc-devs/pymc/blob/c76b9b9e74ee6f13993f255dd09d9290b882d6ab/pymc/distributions/timeseries.py#L239

A similar question arises in the AR, where we again reuse the innovation noise `sigma` for the uncertainty about the initial location of the series.

https://github.com/pymc-devs/pymc/blob/c76b9b9e74ee6f13993f255dd09d9290b882d6ab/pymc/distributions/timeseries.py#L429

### PR DESCRIPTION
**Pull request by ricardoV94**
Sorry @AlexAndorra, I went ahead and opened a PR for this. Tests likely to fail for now :)

Closes #5744

### CODE REVIEW

File: pymc/distributions/timeseries.py [hunk 1]
```diff
@@ -170,16 +172,16 @@ def rng_fn(
...
         rng: np.random.RandomState
            Numpy random number generator
-        mu: array_like
+        mu: array_like of float
            Random walk mean
-        sigma: np.ndarray
+        sigma: array_like of float
             Standard deviation of innovation (sigma > 0)
-        init: float
+        init_dist: array_like of float
```
Comment (twiecki) [inline pymc/distributions/timeseries.py:179]: float? Shouldn't this be a distribution?

File: pymc/distributions/timeseries.py [hunk 2]
```diff
@@ -452,7 +465,13 @@ def dist(
...
                 )
         else:
-            init_dist = Normal.dist(0, 1, size=(*sigma.shape, ar_order))
+            warnings.warn(
+                "Initial distribution not specified, defaulting to "
+                "`Normal.dist(0, 100, shape=...)`. You can specify an init_dist "
+                "manually to suppress this warning.",
+                UserWarning,
+            )
+            init_dist = Normal.dist(0, 100, size=(*sigma.shape, ar_order))
```
Comment (twiecki) [inline pymc/distributions/timeseries.py]: Should use shape.

File: pymc/distributions/timeseries.py (hunk 1 above)
Comment (ricardoV94) [reply pymc/distributions/timeseries.py:179]: In the `rng_fn` we get non-symbolic inputs already

Comment (ricardoV94) [reply pymc/distributions/timeseries.py:179]: In the `rng_fn`, we get the non-symbolic values already

Comment (twiecki) [reply pymc/distributions/timeseries.py:179]: Got it.

File: pymc/distributions/timeseries.py (hunk 2 above)
Comment (ricardoV94) [reply pymc/distributions/timeseries.py]: +1 for picking one or the other.

Changing it

**Review by twiecki — APPROVED** [review]

**Comment by twiecki** [comment]
One down!

**Pull request merged**